pubtools/pulplib/_impl/model/repository/base.py

```diff
@@ -188,6 +190,57 @@ def distributor(self, distributor_id):
         """
         return self._distributors_by_id.get(distributor_id)
 
+    def search_content(self, name=None, arch=None, filename=None, stream=None):
```

Comment:
It shouldn't have the unit-specific fields like name, arch, filename, stream directly built-in to the method signature like this.  It won't scale to support all unit types and it's entirely inconsistent with how the existing search methods work (like searching for repositories).  It should take the criteria object directly, rather than only allowing the method to be used for some specific hardcoded cases.

type_ids also shouldn't be hardcoded in the method body.  This should work with all the unit types supported by the library, so it should work for ISO units too.  Plus, there's a performance penalty which may be significant when you search for type_ids you don't need.  As I understand it, each type_id results in an additional mongo search (if the repo contains any units of that type).  I think the caller needs to specify which types of unit they're interested in.